libopae/src/reconf.c

```diff
@@ -134,6 +135,76 @@ static fpga_result validate_bitstream(fpga_handle handle,
 }
 
 
+// determine if ports for fpga are busy
+static fpga_result port_busy(fpga_handle handle)
+{
+	fpga_result result                = FPGA_OK;
+	struct _fpga_handle *_handle      = (struct _fpga_handle *)handle;
+	fpga_token token;
+	fpga_properties props;
+	fpga_handle accelerator_handle;
+	uint32_t matches = 0;
+
+	if (_handle == NULL) {
+		FPGA_ERR("Invalid handle");
+		return FPGA_INVALID_PARAM;
+	}
+
+	if (_handle->token == NULL) {
+		FPGA_ERR("Invalid token within handle");
+		return FPGA_INVALID_PARAM;
+	}
+
+	result = fpgaGetProperties(NULL, &props);
+	if (result != FPGA_OK)
+		return result;
+
+	result = fpgaPropertiesSetParent(props, _handle->token);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Error setting parent in properties.");
+		goto free_props;
+	}
+
+	// TODO: Use slot number as part of filter
+	//       We only want to query for accelerators for the
+	//       slot being reconfigured
+	result = fpgaEnumerate(&props, 1, &token, 1, &matches);
+	if (result != FPGA_OK) {
+		FPGA_ERR("Error enumerating while reconfiguring");
+		goto free_props;
+	}
+
+	if (0 == matches) {
+		FPGA_ERR("No accelerators found while reconfiguring.");
```

Comment:
And I think we usually don't use a `.` at the end of `FPGA_ERR()` messages, but I'd need to check other code.